src/components/ProgressIndicator/ProgressIndicator.js

```diff
@@ -9,49 +9,98 @@ import PropTypes from 'prop-types';
 
 const { prefix } = settings;
 
 export const ProgressStep = ({ ...props }) => {
-  const { label, description, className, current, complete } = props;
+  const {
+    label,
+    description,
+    className,
+    current,
+    complete,
+    invalid,
+    secondaryLabel,
+    disabled,
+    renderTooltip,
+  } = props;
 
   const classes = classnames({
     [`${prefix}--progress-step`]: true,
     [`${prefix}--progress-step--current`]: current,
     [`${prefix}--progress-step--complete`]: complete,
     [`${prefix}--progress-step--incomplete`]: !complete && !current,
+    [`${prefix}--progress-step--disabled`]: disabled,
     [className]: className,
   });
 
-  const currentSvg = current && (
-    <svg>
-      <circle cx="12" cy="12" r="12" />
-      <circle cx="12" cy="12" r="6" />
-      <title>{description}</title>
-    </svg>
-  );
-
-  const completeSvg = complete && (
-    <svg width="16" height="16" viewBox="0 0 16 16">
-      <title>{description}</title>
-      <g fillRule="nonzero">
-        <path d="M8 15A7 7 0 1 0 8 1a7 7 0 0 0 0 14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z" />
-        <path d="M11.646 5.146l.708.708-5.604 5.603-3.104-3.103.708-.708 2.396 2.397z" />
-      </g>
-    </svg>
-  );
-
-  const incompleteSvg = !complete && (
-    <svg>
-      <title>{description}</title>
-      <circle cx="12" cy="12" r="12" />
-    </svg>
-  );
+  const currentSvg =
+    current &&
+    (componentsX ? (
+      <svg>
+        <path d="M 7, 7 m -7, 0 a 7,7 0 1,0 14,0 a 7,7 0 1,0 -14,0" />
+      </svg>
+    ) : (
+      <svg>
+        <circle cx="12" cy="12" r="12" />
+        <circle cx="12" cy="12" r="6" />
+        <title>{description}</title>
+      </svg>
+    ));
+
+  const completeSvg =
+    complete &&
+    (componentsX ? (
+      <CheckmarkOutline16 aria-label={description} role="img" />
+    ) : (
+      <svg width="16" height="16" viewBox="0 0 16 16">
+        <title>{description}</title>
+        <g fillRule="nonzero">
+          <path d="M8 15A7 7 0 1 0 8 1a7 7 0 0 0 0 14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z" />
+          <path d="M11.646 5.146l.708.708-5.604 5.603-3.104-3.103.708-.708 2.396 2.397z" />
+        </g>
+      </svg>
+    ));
+  const incompleteSvg = (() => {
+    if (complete) {
+      return null;
+    }
+    if (componentsX) {
+      if (invalid) {
+        return <Warning16 className={`${prefix}--progress__warning`} />;
+      }
+      return (
+        <svg>
+          <path d="M8 1C4.1 1 1 4.1 1 8s3.1 7 7 7 7-3.1 7-7-3.1-7-7-7zm0 13c-3.3 0-6-2.7-6-6s2.7-6 6-6 6 2.7 6 6-2.7 6-6 6z" />
+        </svg>
+      );
+    }
+    return (
+      <svg>
+        <title>{description}</title>
+        <circle cx="12" cy="12" r="12" />
+      </svg>
+    );
+  })();
 
   return (
     <li className={classes}>
       {currentSvg || completeSvg || incompleteSvg}
-      <p className={`${prefix}--progress-label`}>{label}</p>
+      {componentsX && renderTooltip ? (
+        renderTooltip()
```

Comment:
* Given `renderTooltip()` is a new feature, looks that we don't need to role it off for `v10`?
* Can we set default render code as the default prop?
* Can we support stateful here as well...?

Imagining something like below, wondering if it makes sense to everybody:

```diff
--- a/src/components/ProgressIndicator/ProgressIndicator.js
+++ b/src/components/ProgressIndicator/ProgressIndicator.js
@@ -17,6 +17,8 @@ import { componentsX } from '../../internal/FeatureFlags';
 
 const { prefix } = settings;
 
+const defaultRenderLabel = props => <p {...props} />;
+
 export const ProgressStep = ({ ...props }) => {
   const {
     label,
@@ -27,7 +29,7 @@ export const ProgressStep = ({ ...props }) => {
     invalid,
     secondaryLabel,
     disabled,
-    renderTooltip,
+    renderLabel: ProgressStepLabel,
   } = props;
 
   const classes = classnames({
@@ -91,11 +93,7 @@ export const ProgressStep = ({ ...props }) => {
   return (
     <li className={classes}>
       {currentSvg || completeSvg || incompleteSvg}
-      {componentsX && renderTooltip ? (
-        renderTooltip()
-      ) : (
-        <p className={`${prefix}--progress-label`}>{label}</p>
-      )}
+      <ProgressStepLabel className={`${prefix}--progress-label`}>{label}</ProgressStepLabel>
       {componentsX &&
       secondaryLabel !== null &&
       secondaryLabel !== undefined ? (
@@ -146,7 +144,7 @@ ProgressStep.propTypes = {
    * An optional parameter to allow for overflow content to be rendered in a
    * tooltip.
    */
-  renderTooltip: PropTypes.function,
+  renderLabel: PropTypes.function,
 
   /**
    * Provide the props that describe a progress step tooltip
@@ -164,6 +162,10 @@ ProgressStep.propTypes = {
   tooltipId: PropTypes.string,
 };
 
+ProgressStep.defaultProps = {
+  renderLabel: defaultRenderLabel,
+};
+
 export class ProgressIndicator extends Component {
   state = {}; 
```

Reply:
makes sense, thanks for the suggested changes @asudoh